cmd/kube-proxy/app/server.go

```diff
@@ -126,15 +131,25 @@ func NewProxyConfig() *ProxyServerConfig {
 
 func NewProxyServer(
 	config *ProxyServerConfig,
+	client *kubeclient.Client,
+	endpointsConfig *proxyconfig.EndpointsConfig,
+	endpointsHandler proxyconfig.EndpointsConfigHandler,
 	iptInterface utiliptables.Interface,
+	oomAdjuster *oom.OOMAdjuster,
 	proxier proxy.ProxyProvider,
 	recorder record.EventRecorder,
+	serviceConfig *proxyconfig.ServiceConfig,
 ) (*ProxyServer, error) {
 	return &ProxyServer{
-		Config:       config,
-		IptInterface: iptInterface,
-		Proxier:      proxier,
-		Recorder:     recorder,
+		Client:           client,
+		Config:           config,
+		EndpointsConfig:  endpointsConfig,
```

Comment:
Probably quite a few rebases. The problem are in dependencies of various parts in the proxy, IIRC lots of stuff depends on a lot of other stuff, which is not directly injectable, but created on requests. This was easiest, yet dirtest, way to allow injection of various things. I'll look into cleaning it up tomorrow (today I'm on an external event:/).